### PR TITLE
Add unit tests for javascript

### PR DESCRIPTION
**Pull request by PierreRambaud**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Add missing unit tests for javascript
| Type?             | improvement
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | CI is green and correctly executed
| Possible impacts? | Please indicate what parts of the software we need to check to make sure everything is alright.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/24927)

### CODE REVIEW

**Review by NeOMakinG — DISMISSED** [review]

File: .github/workflows/js.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    runs-on: ubuntu-latest
+    name: Run unit tests
+    strategy:
+      matrix:
+        js: ['10', '11', '12', '13', '14', '15']
+    steps:
+      - name: Setup Node
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.php }}
```
Comment (Progi1984) [inline .github/workflows/js.yml]:
```suggestion
          node-version: ${{ matrix.js }}
```

File: .github/workflows/js.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        uses: actions/cache@v2
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+               ${{ runner.os }}-node-
+
+      - name: Run tests
+        run: |
+          npm install && npm run test
```
Comment (Progi1984) [inline .github/workflows/js.yml]:
```suggestion
        run: npm install && npm run test
```

File: .github/workflows/js.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        with:
+          node-version: ${{ matrix.php }}
+
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Get Composer Cache Directory
+        id: composer-cache
+        run: |
+          echo "::set-output name=dir::$(composer config cache-files-dir)"
```
Comment (Progi1984) [inline .github/workflows/js.yml]:
```suggestion
        run: echo "::set-output name=dir::$(composer config cache-files-dir)"
```

**Review by Progi1984 — COMMENTED** [review]
Some feedbacks

File: .github/workflows/js.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        with:
+          node-version: ${{ matrix.js }}
+
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Get Composer Cache Directory
+        id: composer-cache
+        run: echo "::set-output name=dir::$(composer config cache-files-dir)"
+
```
Comment (Progi1984) [inline .github/workflows/js.yml]: Remove Cache (it's for composer).

Comment (Progi1984) [reply .github/workflows/js.yml]:
```suggestion
```

File: .github/workflows/js.yml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+      - name: Cache Node Directory
+        uses: actions/cache@v2
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+               ${{ runner.os }}-node-
```
Comment (Progi1984) [inline .github/workflows/js.yml]:
```suggestion
          restore-keys: ${{ runner.os }}-node-
```

**Review by Progi1984 — COMMENTED** [review]
Last feedback. Sorry @PierreRambaud

**Review by Progi1984 — DISMISSED** [review]

**Review by Progi1984 — APPROVED** [review]

**Review by NeOMakinG — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks @PierreRambaud

**Pull request merged**